crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -452,7 +493,11 @@ where
                     &filter_params,
                     chain_info.into(),
                     &receipts,
-                    false,
+                    self.reorged_blocks
+                        .lock()
+                        .await
+                        .get(&block.number)
+                        .is_some_and(|hash| *hash == block.hash()),
```

Comment:
this is not exactly what we want here.

reorg support is basically only required for the `filter_changes`

the reorg info is needed so that we emit a previously emitted Log again but this time with `removed: true`

so what watch blocks should do is:
1. detect reorged blocks
2. find active filters that are affected by this reorg, these are filters that emitted logs of a now reorged block